### PR TITLE
Reduce janitor log spam.

### PR DESCRIPTION
**Pull request by amwat**
Janitor ends up spamming this log when there is nothing to clean, change to warning instead.

/cc @krzyzacy @fejta

### CODE REVIEW

File: boskos/boskos.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func handleAcquire(r *ranch.Ranch) http.HandlerFunc {
 		resource, err := r.Acquire(rtype, state, dest, owner)
 
 		if err != nil {
-			logrus.WithError(err).Errorf("No available resource")
+			logrus.WithError(err).Warningf("No available resource")
```
Comment (krzyzacy) [inline boskos/boskos.go]: I think you can check err type here (https://github.com/kubernetes/test-infra/blob/master/boskos/ranch/ranch.go#L46-L82) and warn if it's a not found error

Comment (amwat) [reply boskos/boskos.go]: done

File: boskos/boskos.go [hunk 2]
```diff
@@ -155,7 +155,11 @@ func handleAcquire(r *ranch.Ranch) http.HandlerFunc {
 		resource, err := r.Acquire(rtype, state, dest, owner)
 
 		if err != nil {
-			logrus.WithError(err).Errorf("No available resource")
+			if err2, ok := err.(ranch.ResourceNotFound); ok {
+				logrus.WithError(err2).Warningf("No available resource")
```
Comment (fejta) [inline boskos/boskos.go]: Can we distinguish between "I'll never have anything to do" and "I'm already done?" The latter seems like it should not be a warning

Comment (krzyzacy) [reply boskos/boskos.go]: I think from server side we cannot identify the difference between `well I'll just do nothing if I don't get a resource` and `Nooo I absolutely need this resource or I'll panic`... We can add an additional param to identify the calls from client side.

**Comment by amwat** [comment]
Do we still need this ?

**Comment by fejta** [comment]
Depends, I would still like it to not error or warn on non-problematic scenarios. So yes?

On the other hand this PR doesn't accomplish that right now.

**Comment by amwat** [comment]
To Sen's point I think it's hard to determine on the boskos server side whether "not finding a resource" is an noteworthy thing to log or not. The janitor client does not care if we were unable to find a resource, on the other hand stuff like e2e tests fail if they can't find a resource. We would wither have to add something like `clientName` and hard code "ignore logging if `client == janitor`", or we could drop this logging altogether and leave it to the client to log? wdyt @krzyzacy

**Comment by krzyzacy** [comment]
Or we should make everyone use AcquireWait, instead of only Acquire once

**Comment by amwat** [comment]
AcquireWait will only work for the boskos go client, not for the raw REST API (e.g. curl) 

Anyway, closing this PR since special casing the log line is not the right solution. 

/close

**Comment by krzyzacy** [comment]
@amwat boskosctl supports that natively, so we can might as well use that instead of curl :-)

**Pull request closed without merging**